src/main/java/com/rocketden/main/dto/problem/CreateProblemRequest.java

```diff
@@ -13,6 +13,7 @@
 public class CreateProblemRequest {
     private String name;
     private String description;
+    private Boolean approval;
```

Comment:
I just realized that if we include `approval` in the `CreateProblemRequest`, we break one of the invariants that was set in this PR - you added a check in the `editProblem` method that the user cannot set the problem to be approved if there are no test cases. However, through creating the problem, we currently have it set up such that you cannot add any test cases until it's actually created, after which you can. You can see that reflected on our UI in the image below:

<img width="1262" alt="Create Problem Page" src="https://user-images.githubusercontent.com/7987279/113495218-d2f8cc00-94a4-11eb-82c3-ab513b39d114.png">

Since it'd be a long fix, I propose that we just include the `approval` boolean for the `editProblem` endpoint. Sorry @alankbi and I didn't identify this earlier! We could enable the ability to add test cases when creating the problem, but that would take a lot more work and we had reasons for preventing this in the first place (we wanted people to first verify that they had the problem inputs set).

Reply:
Got it, I just delete that `Boolean approval` line I added in `CreateProblemRequest right, or is there more to this fix?

Reply:
It might be slightly more challenging than that because in the `createSingleProblem`-type methods (the ones in the test files), it's currently just setting approval to true on creation of the problem, so in order to still have that feature (which allows the tests to pass), you'd need to add an additional edit problem request to set the approval to true.

Reply:
I don't think you've made this change yet @sean-xcamp - the list I think is this:
1. Remove the `approval` field in `CreateProblemRequest.java `.
2. Remove all the relevant approval code for `createProblem` in `ProblemService.java`. (You can look at exactly this link - [https://github.com/rocketden/main/pull/131/files](https://github.com/rocketden/main/pull/131/files) - to simply remove the changes one-by-one.)
3. Remove the approval setting in `SocketTestMethods.java`.

There may be others but that's all I could find, I suggest looking at the same "Files Changed" link [here](https://github.com/rocketden/main/pull/131/files) to revert the relevant changes.